### PR TITLE
layers: Fix pipeline rendering info validation

### PR DESCRIPTION
**Pull request by ziga-lunarg**
Move `VkPipelineRenderingCreateInfo` out of the fragment shader state condition and into a separate function

### CODE REVIEW

File: layers/stateless/sl_pipeline.cpp [hunk 1]
```diff
@@ -441,58 +487,14 @@ bool StatelessValidation::manual_PreCallValidateCreateGraphicsPipelines(
...
-                    }
-                }
+            const auto rendering_struct = vku::FindStructInPNextChain<VkPipelineRenderingCreateInfo>(create_info.pNext);
+            if (rendering_struct) {
+                color_attachment_count = rendering_struct->colorAttachmentCount;
+                skip |= ValidatePipelineRenderingCreateInfo(*rendering_struct, create_info_loc);
+            }
 
+            // Pipeline has fragment output state
+            if (create_info.pColorBlendState && create_info.pMultisampleState) {
```
Comment (spencer-lunarg) [inline layers/stateless/sl_pipeline.cpp:497]: looked at spec, `VkAttachmentSampleCountInfoAMD` itself works like `VkPipelineRenderingCreateInfo`

so basically asking to remove

```patch
-             // Pipeline has fragment output state
-            if (create_info.pColorBlendState && create_info.pMultisampleState) {
```

as this is not a requirement for that struct neither

**Review by spencer-lunarg — APPROVED** [review]
LGTM with one nit

File: layers/stateless/sl_pipeline.cpp (hunk 1 above)
Comment (ziga-lunarg) [reply layers/stateless/sl_pipeline.cpp]: Turns out this might be needed for libraries

File: layers/stateless/sl_pipeline.cpp [hunk 2]
```diff
@@ -168,56 +214,13 @@ bool StatelessValidation::manual_PreCallValidateCreateGraphicsPipelines(
         uint32_t color_attachment_count = 0;
 
         if (!create_info.renderPass) {
-            if (create_info.pColorBlendState && create_info.pMultisampleState) {
+            // Pipeline has fragment output state
+            if ((create_info.flags & VK_PIPELINE_CREATE_LIBRARY_BIT_KHR) == 0 ||
```
Comment (spencer-lunarg) [inline layers/stateless/sl_pipeline.cpp:218]: aww, the issue is with GPL, you do need the "full" version of the state... this is all still a mess, but glad to have a test and working case for now

**Pull request merged**